docs/windows-design.md

```diff
@@ -0,0 +1,249 @@
+OVS is working as the extension of the external Hyper-V switch, which is bound with the HNS 
+Network. Hence, the packets that are sent from/to the containers could be processed by OVS.
+The network adapter used in the HNS Network is also added into the OVS bridge as the uplink
+interface. An interface for the OVS bridge is created, and the original networking
+configuration(e.g., IP, MAC and route entries) on the host network adapter are moved to
+this new interface. Some extra OpenFlow entries are needed to ensure the host traffic can be
+forwarded correctly.
+
+<img src="/docs/assets/hns_integration.svg" width="600" alt="HNS Integration">
+
+SNAT based on OpenFlow is needed to make sure the containers could access the external address.
+The SNATed address is using the IP configured on the OVS bridge. Some assisted OpenFlow entries are
+installed to help identify and forward the external traffic.
+
+kube-proxy is running on each Windows Node to provide Kubernetes Service. The Service traffic is
+output to the host from gw0, and captured by kube-proxy. kube-proxy sets up a new connection to
+the backend endpoint, and forwarded the request using this new connection. The forwarded request
+enters OVS pipeline again and output to the actual Pod. To be compatible with OVS, kube-proxy on
+Windows must be configured to run in the **userspace** mode, and a specific network adapter is 
+required to be provided on which the Service IP address is configured.
+
+### HNS Network configuration
+
+HNS Network is created at Antrea Agent initialization phase, and it should be created before creating 
+OVS bridge. This is because OVS is working as the Hyper-V Switch Extension, and the ovs-vswitchd process
+cannot run until the Hyper-V Switch is create and the OVS Extension is enabled on this Switch. 
+
+When creating the HNS Network, the local subnet CIDR and the uplink network adapter are required.
+Antrea Agent finds the network adapter from the Windows host using the Node internal IP as a filter,
+and retrieves the local Subnet CIDR from Node spec. 
+
+After the HNS Network is created, OVS extension should be enabled at once.
+
+### Container network configuration
+
+[**host-local**](https://github.com/containernetworking/plugins/tree/master/plugins/ipam/host-local)
+plugin is used to provide IPAM for containers, and the address is allocated from the subnet CIDR
+configured on the HNS Network. 
+
+Windows HNS Endpoint is leveraged as the vNIC for each container. A single HNSEndpoint with the
+IP allocated from IPAM plugin is created for each Pod. The HNSEndpoint should bw attached on all
+containers in the same Pod to ensure thw network configuration could be correctly accessed(this
+operation is to make sure DNS configuration is readable from all containers).
+ 
+One OVS port with the same name as the HNS Endpoint is also needed, which is helping to control
+the container traffic with OpenFlow rules. OpenFlow entries are installed to implement connectivity
```

Comment:
s/which is helping to control the container traffic with OpenFlow rules/in order to handle container traffic with OpenFlow rules